### DISCUSSION TITLE
[Feature Request] Restore existing chat threads on connecting

### DISCUSSION THREAD

**Discussion by MuckT** [WalletConnect/walletconnect-monorepo#2427] [Developer Support] [answered]
### Description

Currently when register my address with chat it fails to restore any of my previous sessions activities. As a user I would expect this behavior to be default. Using Discord as an example it would be strange to log in to my discord account and re-invite friend and start chat threads from scratch.

While some of this might be handled in the dapp or wallet state, without the ability to restore threads for a wallet from the keyserver users who get a new device would still lose all their chat invites and thread and messages.

**Comment by Elyniss** [comment, marked as answer]
Hey! I have good news! This feature is already being worked on and is one of our biggest priorities at the moment.
